src/lsm/cache_map.zig

```diff
@@ -31,39 +31,33 @@ pub fn CacheMapType(
     comptime tombstone_from_key: fn (Key) callconv(.Inline) Value,
     comptime tombstone: fn (*const Value) callconv(.Inline) bool,
 ) type {
-    const _Cache = SetAssociativeCacheType(
-        Key,
-        Value,
-        key_from_value,
-        hash_from_key,
-        .{},
-    );
-
-    const HashMapContextValue = struct {
-        const Self = @This();
-
-        pub inline fn eql(_: Self, a: Value, b: Value) bool {
-            return key_from_value(&a) == key_from_value(&b);
-        }
-
-        pub inline fn hash(_: Self, value: Value) u64 {
-            return stdx.hash_inline(key_from_value(&value));
-        }
-    };
-
-    const map_load_percentage_max = 50;
-    const _Map = std.HashMapUnmanaged(
-        Value,
-        void,
-        HashMapContextValue,
-        map_load_percentage_max,
-    );
-
     return struct {
-        const Self = @This();
+        const CacheMap = @This();
+
+        const map_load_percentage_max = 50;
+
+        pub const Cache = SetAssociativeCacheType(
+            Key,
+            Value,
+            key_from_value,
+            hash_from_key,
+            .{},
+        );
+
+        pub const Map = std.HashMapUnmanaged(
+            Key,
+            Value,
```

Comment:
Hm, this still doesn't feel quite right: we do duplicate data in memory. 56 megs is indeed not much in term of RAM, _but_ it is quite a bit in terms of CPU cache, and we already have values in CPU cache. 

Let me take closer look at the get and remove changes here...

Reply:
Since the benchmark didn't catch anything meaningful difference, let's stick with the memory we saved with Value/Void (the last commit can be reverted to compare both cases).

Reply:
:+1: I mostly want to make sure that I am _not_ missing anything, and that storing value is indeed redundant, abstractly speakin!

Reply:
FYI, [historically](https://github.com/tigerbeetle/tigerbeetle/commit/9c8d816577184558db7ad25eaa03dad5861948a9#diff-be4726c0a7b96915c03bad6692df47aa2c623e8066d22367a5e9c1755857dfe8R363) we always used Value/Void

```zig
 const PrefetchObjects = std.HashMapUnmanaged(Object, void, PrefetchObjectsContext, 70);
```

Reply:
I'd argue that `CacheMap` _is_ that  common API wrapper 🤔